### PR TITLE
Config v2 tests

### PR DESCRIPTION
**Pull request by andreamlin**
- Update config_v2.proto with the new fields in config.proto
- Add the reduced library_gapic_v2.yaml
- Add a baseline Java test for GAPIC config v2; the goal is to iteratively reduce the diff between this file and the other java_library.baseline created from GAPIC config v1

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

**Comment by andreamlin** [comment]
More tests are coming in another PR

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+    package_name: Library::V1
+  # Use default PHP package name.
+  nodejs:
+    package_name: library.v1
+    domain_layer_location: google-cloud
+  csharp:
+    package_name: Google.Example.Library.V1
+interfaces:
+  - name: google.example.library.v1.MyProto
+    retry_codes_def:
```
Comment (michaelbausor) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml]: Is this necessary? We should be able to omit the retry codes from v2 config if they just specify the defaults.

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml [hunk 2]
```diff
@@ -0,0 +1,693 @@
...
+            common_resource_name: com.google.cloud.ProjectName
+    smoke_test:
+      method: UpdateBook
+      init_fields:
+        - name%shelf_id="testShelf-$RANDOM"
+        - name%book_id=$PROJECT_ID
+        - book . rating=GOOD
+    methods:
+      - name: CreateShelf
+        retry_codes_name: non_idempotent
```
Comment (michaelbausor) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml]: Can these be inferred from http annotation? We should have a mix of most being inferred, but some being overridden. WDYT?

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml (hunk 1 above)
Comment (andreamlin) [reply src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml]: Done.

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml (hunk 2 above)
Comment (andreamlin) [reply src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml]: Ahh I got lazy
Removed the retry codes names that are the same as the inferred value.

**Comment by andreamlin** [comment]
removed redundant config values
PTAL

**Comment by andreamlin** [comment]
omg i put the requested changes in another branch instead of this one
will push those v soon

**Comment by michaelbausor** [comment]
Heh, my fault also for bad LGTM, whoops :-)

**Pull request merged**